Symfony/CS/Console/Command/SelfUpdateCommand.php

```diff
@@ -55,33 +55,32 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
-        $currentVersion = explode('-', $this->getApplication()->getVersion());
-        $currentVersion = $currentVersion[0]; // ignore index #1 if exists (drop non-stable versions like `-DEV`)
-        $currentVersion = explode('.', $currentVersion);
-        if (!isset($currentVersion[2])) {
-            $currentVersion[2] = 0; // fill patch version if missing
-        }
+        $remoteTag = $this->getRemoteTag();
 
-        list($major, $minor, $patch) = $this->findBestVersion($currentVersion[0], $currentVersion[1], $currentVersion[2]);
+        if (null === $remoteTag) {
+            $output->writeln('<error>Unable to determine newest version.</error>');
 
-        if ($this->getApplication()->getVersion() === $this->buildVersionString($major, $minor, $patch)) {
+            return;
+        }
+
+        if ('v' . $this->getApplication()->getVersion() === $remoteTag) {
             $output->writeln('<info>php-cs-fixer is already up to date.</info>');
 
             return;
         }
 
-        $remoteFilename = $this->buildVersionFileUrl($major, $minor, $patch);
+        $remoteFilename = $this->buildVersionFileUrl($remoteTag);
         $localFilename = $_SERVER['argv'][0];
         $tempFilename = basename($localFilename, '.phar').'-tmp.phar';
 
-        if (false === @file_get_contents($remoteFilename)) {
-            $output->writeln('<error>Unable to download new versions from the server.</error>');
+        try {
+            $copyResult = @copy($remoteFilename, $tempFilename);
+            if (false === $copyResult) {
+                $output->writeln('<error>Unable to download new versions from the server.</error>');
 
-            return 1;
-        }
+                return 1;
+            }
 
-        try {
-            copy($remoteFilename, $tempFilename);
```

Comment:
We could. that that would still mean we send second HTTP request.
Now we send only one request, still reporting if file is not there.